src/persist/src/location.rs

```diff
@@ -931,6 +967,49 @@ pub mod tests {
         // We can observe the a recent value on successful update.
         assert_eq!(consensus.head(deadline, key).await, Ok(Some(state.clone())));
 
+        // Can scan a key that has data without providing a lower bound on the
+        // sequence number.
+        assert_eq!(
+            consensus.scan(deadline, key, None).await,
+            Ok(vec![state.clone()])
+        );
+
+        // Can scan a key that has data with a lower bound sequence number < head.
+        assert_eq!(
+            consensus.scan(deadline, key, Some(SeqNo(0))).await,
+            Ok(vec![state.clone()])
+        );
+
+        // Can scan a key that has data with a lower bound sequence number == head.
+        assert_eq!(
+            consensus.scan(deadline, key, Some(SeqNo(5))).await,
+            Ok(vec![state.clone()])
+        );
+
+        // Can scan a key that has data with a lower bound sequence number > head, but we won't see
```

Comment:
this feels to me like it should be an error, no? all the cases for scan I can think of derive the scan start from observing something, so it's pretty unexpected to get one that's higher than current

Reply:
done.